subprojects/plugins/src/main/java/org/gradle/api/plugins/WarPlugin.java

```diff
@@ -88,24 +106,34 @@ public void apply(final Project project) {
 
         PublishArtifact warArtifact = new LazyPublishArtifact(war, ((ProjectInternal) project).getFileResolver(), ((ProjectInternal) project).getTaskDependencyFactory());
         project.getExtensions().getByType(DefaultArtifactPublicationSet.class).addCandidate(warArtifact);
-        configureConfigurations(project.getConfigurations());
+        configureConfigurations((RoleBasedConfigurationContainerInternal) project.getConfigurations());
         configureComponent(project, warArtifact);
     }
 
-    public void configureConfigurations(ConfigurationContainer configurationContainer) {
-        Configuration providedCompileConfiguration = configurationContainer.create(PROVIDED_COMPILE_CONFIGURATION_NAME).setVisible(false).
-            setDescription("Additional compile classpath for libraries that should not be part of the WAR archive.");
-        providedCompileConfiguration.setCanBeConsumed(false);
+    private void configureConfigurations(RoleBasedConfigurationContainerInternal configurationContainer) {
+        Configuration providedCompileConfiguration = configurationContainer.createWithRole(PROVIDED_COMPILE_CONFIGURATION_NAME, ConfigurationRoles.INTENDED_BUCKET)
```

Comment:
Switching these right away to an `INTENDED_BUCKET` without a deprecation cycle is going to be a breaking change. We need something like a `DEPRECATED_BUCKET` for 8.1 which deprecates for resolving and consuming, and then we can switch this over to an `INTENDED_BUCKET` in 9.0

Reply:
It would have to be a custom configuration: resolvable + declarable but with resolution deprecated, since consuming is already banned.

From the wording of the issue, I understood this to mean go ahead and make the breaking change.  The new extending configurations being used for resolution could be a breaking change anyway.

I don't know if there's much benefit trying to do this so cautiously.  If you're manually resolving a `visible=false` configuration, that should be a hint you're doing something wrong.  Let's just make the full change and add a note to the upgrade guide.

Reply:
Targeting 9.0 now, so this is a moot point.